cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -97,22 +97,36 @@ void reader::impl::decode_page_data(read_mode mode, size_t skip_rows, size_t num
                              _stream);
     }
 
+    // column string sizes for this subpass
     col_string_sizes = calculate_page_string_offsets();
 
+    // ensure cumulative column string sizes have been initialized
+    if (pass.cumulative_col_string_sizes.empty()) {
+      pass.cumulative_col_string_sizes.resize(_input_columns.size(), 0);
+    }
+
+    // add to cumulative column string sizes from this subpass
+    std::transform(pass.cumulative_col_string_sizes.begin(),
+                   pass.cumulative_col_string_sizes.end(),
+                   col_string_sizes.begin(),
+                   pass.cumulative_col_string_sizes.begin(),
```

Comment:
Could we avoid the resize above by using a back_inserter here?

Reply:
Technically yes, based on conditional; back_insert if empty else add vectors. Though, `resize` happens only once per reader `pass` based on the same predicate.

Reply:
> back_insert if empty else add vectors

That's definitely not better than the current code. Resolving :)